common/subtitle-coloring/subtitle-coloring.ts

```diff
@@ -0,0 +1,790 @@
+                    newSubtitles.push(this._subtitles[index]); // Process all uncollected subtitles even if not in buffer
+                }
+                subtitles = newSubtitles;
+            }
+
+            await inBatches(
+                subtitles,
+                async (batch) => {
+                    await Promise.all(
+                        batch.map(async ({ index, text, track }) => {
+                            if (this.shouldCancelBuild) return;
+                            try {
+                                this.colorCacheBuildingCurrentIndexes.add(index);
+                                const ts = this.trackStates[track];
+                                if (!this._dictionaryTrackEnabled(ts.dt)) return;
+                                const cachedRichText = this._subtitles[index].richText;
+                                if (this._colorCacheValid(cachedRichText, index)) return;
+                                const richText = await this._colorizeText({ text, index, ts });
+                                if (cachedRichText === richText) return;
+                                if (this.shouldCancelBuild) return;
+                                const updatedSubtitles: RichSubtitleModel[] = [];
+                                if (richText) {
+                                    this._subtitles[index].richText = richText;
+                                    updatedSubtitles.push(this._subtitles[index]);
+                                }
+                                this.subtitleColorsUpdated(updatedSubtitles);
+                            } finally {
+                                this.colorCacheBuildingCurrentIndexes.delete(index);
+                            }
+                        })
+                    );
+                },
+                { batchSize: TOKEN_CACHE_BATCH_SIZE }
+            );
+            if (this.shouldCancelBuild) {
+                buildWasCancelled = true;
+                uncollectedWasRefreshed = false;
+            }
+        } finally {
+            if (this.tokenRequestFailed) {
+                this.tokenRequestFailed = false;
+                this.trackStates.forEach((ts) => (ts.yomitanHealthy = false));
+                this.anki = undefined;
+            }
+            if (uncollectedWasRefreshed) this.uncollectedNeedsRefresh = false;
+            this.shouldCancelBuild = false;
+            this.colorCacheBuilding = false;
+        }
+        return !buildWasCancelled;
+    }
+
+    private async _colorizeText(options: { text: string; index: number; ts: TrackState }): Promise<string | undefined> {
+        const { text, index, ts } = options;
+        try {
+            if (!this.anki) throw new Error('Anki not initialized');
+            if (!ts.yomitanHealthy) throw new Error(`Yomitan not initialized for Track${ts.track + 1}`);
+            if (!ts.dt.dictionaryAnkiWordFields.length && !ts.dt.dictionaryAnkiSentenceFields.length) {
```

Comment:
Should this check be made higher in the stack? If we're treating this as an error maybe we should also warn users in the settings form if they enable coloring without specifying these fields.